starter_code/src/movies.js

```diff
@@ -1,20 +1,71 @@
 /* eslint no-restricted-globals: 'off' */
-// Turn duration of the movies from hours to minutes 
+// Turn duration of the movies from hours to minutes
 
+// Get the average of all rates with 2 decimals
 
-// Get the average of all rates with 2 decimals 
+function ratesAverage(array) {
+  var averageRate = array.reduce(function(sum, oneMovie) {
+    var totalRate = sum + parseInt(oneMovie.rate);
 
+    return totalRate;
+  }, 0);
+
+  var stringResult = (averageRate / array.length).toFixed(2);
+  return parseFloat(stringResult);
+}
+
+console.log(ratesAverage(movies));
 
 // Get the average of Drama Movies
 
+var dramaMovies = movies.filter(function(oneMovie) {
+  if (oneMovie.genre.indexOf("Drama") >= 0) {
+    return oneMovie;
+  }
+});
+
+function dramaMoviesRate(array) {
```

Comment:
To pass the final test for this function you need to add 

``` 
 if (dramaMovies.length === 0) {
    return undefined;
  }
```

This checks to see if there are any drama movies, and if not, returns undefined